### PR TITLE
Convert `dd` commands to rzshell

### PR DESCRIPTION
**Pull request by XVilka**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

* Convert `dd` command to rzshell
* Listing descriptors moved from `dd` to `ddl`
* Added JSON and table modes to `ddl`
* Remove unimplemented `dH` command
* Remove `ddt` command
* Move `date` command to the "shell" category

**Test plan**

- CI is green
- Try `dd` commands
- Try the `date` command

**Closing issues**

Partially addresses https://github.com/rizinorg/rizin/issues/1342

### CODE REVIEW

File: librz/core/cmd_descs/cmd_debug.yaml [hunk 1]
```diff
@@ -312,6 +312,67 @@ commands:
...
+          - name: offset
+            type: RZ_CMD_ARG_TYPE_RZNUM
+      - name: ddd
+        summary: Duplicate <fd1> to <fd2>
+        cname: cmd_debug_descriptor_dup
+        args:
+          - name: fd1
+            type: RZ_CMD_ARG_TYPE_RZNUM
+          - name: fd2
+            type: RZ_CMD_ARG_TYPE_RZNUM
```
Comment (wargio) [inline librz/core/cmd_descs/cmd_debug.yaml]: I belive that all the file descriptors needs to be `RZ_CMD_ARG_TYPE_NUM` and not `RZ_CMD_ARG_TYPE_RZNUM` and i would rename `fd1` to `fd_src` and `fd2` to `fd_dst`

Comment (wargio) [reply librz/core/cmd_descs/cmd_debug.yaml]: the file descriptor comment applies to all the commands.

**Review by wargio — APPROVED** [review]

File: librz/core/cmd_descs/cmd_debug.yaml (hunk 1 above)
Comment (ret2libc) [reply librz/core/cmd_descs/cmd_debug.yaml]: Yeah I think so too.

**Review by ret2libc — APPROVED** [review]
Approved except what mentioned by @wargio .

**Pull request merged**